src/systemctl/systemctl-whoami.c

```diff
@@ -1,32 +1,104 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include "bus-common-errors.h"
 #include "bus-error.h"
 #include "bus-locator.h"
+#include "format-util.h"
+#include "parse-util.h"
+#include "pidref.h"
+#include "process-util.h"
 #include "systemctl.h"
 #include "systemctl-util.h"
 #include "systemctl-whoami.h"
-#include "parse-util.h"
 
-static int lookup_pid(sd_bus *bus, pid_t pid) {
+static bool fallback = false;
```

Comment:
maybe name this "pidfd_fallback"?

Reply:
hmm, you only need this inside of lookup_pid(), hence maybe move it into the function. It's usually how we do this with caches.